### DISCUSSION TITLE
Enum custom validation messages

### DISCUSSION THREAD

**Discussion by Tripsy** [laravel/framework#50248] [Q&A]
I'm not able to add a custom validation message for enum rules.

I have my rule defined as:

`'components.*.component_type' => ['required', Rule::enum(BlueprintComponentType::class)]`

The custom message is defined as:

`
    'custom' => [
        'components.*.component_type' => [
            'enum' => 'The component type field should be ...',
        ],
    ],
`

The output is "The selected components.0.component_type is invalid" which is the value for `validation.enum`

Am I doing something wrong?

**Comment by Tripsy** [comment]
PS: I'm using Laravel 10.45

**Comment by denizgolbas** [comment]
```php
'components.*.component_type' => ['required', new Enum(BlueprintComponentType::class)]
```

**Change like this and add:**

```php
use Illuminate\Validation\Rules\Enum;
```

**Reply by Tripsy** [reply]
This doesn't help. Rule::enum is returning an Enum object anyway. 

I feel that the problem is inside the Laravel class vendor/laravel/framework/src/Illuminate/Validation/Rules/Enum.php method message which is set to return only the lang validation.enum

**Comment by firmanalams** [comment]
For custom error message inside `lang` folder is not implemented yet, try using `FormRequest` instead

if you looked at file `src/Illuminate/Validation/Rules/Enum.php`

```
    public function message()
    {
        $message = $this->validator->getTranslator()->get('validation.enum');

        return $message === 'validation.enum'
            ? ['The selected :attribute is invalid.']
            : $message;
    }
```

the `message()` function is just get translation for `validation.enum`
